src/ids.rs

```diff
@@ -622,8 +622,92 @@ impl<'de> serde::Deserialize<'de> for InvoiceId {
 
 #[cfg(test)]
 mod tests {
+    use std::fmt::{Debug, Display};
+    use std::str::FromStr;
+
+    use serde::de::DeserializeOwned;
+    use serde::{Deserialize, Serialize};
+    use serde_json::json;
+
     use super::*;
 
+    fn assert_ser_de_roundtrip<T>(id: &str)
+    where
+        T: DeserializeOwned + Serialize + FromStr + Display + Debug,
+        <T as FromStr>::Err: Debug,
+    {
+        let parsed_id = T::from_str(id).expect("Could not parse id");
+        let ser = serde_json::to_string(&parsed_id).expect("Could not serialize id");
+        let deser: T = serde_json::from_str(&ser).expect("Could not deserialize id");
+        assert_eq!(deser.to_string(), id.to_string());
+    }
+
+    fn assert_deser_err<T: DeserializeOwned + Debug>(id: &str) {
+        let json_str = format!(r#""{}""#, id);
+        let deser: Result<T, _> = serde_json::from_str(&json_str);
+        assert!(deser.is_err(), "Expected error, got {:?}", deser);
+    }
+
+    #[test]
+    fn test_empty_invoice_id_default() {
+        #[derive(Deserialize)]
+        struct WithInvoiceId {
+            id: InvoiceId,
+        }
+
+        for body in [json!({"id": ""}), json!({})] {
+            let deser: WithInvoiceId = serde_json::from_value(body).expect("Could not deser");
+            assert_eq!(deser.id, InvoiceId::none());
+        }
```

Comment:
The special-cased behavior of `InvoiceId` looks suspicious - do you know what it came from? Seems like using `Option<InvoiceId>` would be a clearer way to handle this `InvoiceId` potentially not being present in data

Reply:
I agree. This fork has not really made any substantial changes to the codegen and there are a number of rough edges. At the very least, an explanation of _why_ this is ok would be important.

I would like to open up a project for overhauling the codegen because (in addition to this issue and no doubt a few more oddities) there is a decent amount of the API surface that is ignored altogether as well, though I worry that it will be hard to do so without causing significant breakages.